starters/vue-starter/hooks/usePagination.js

```diff
@@ -0,0 +1,52 @@
+import { ref, watch } from "vue";
+
+export default function usePagination({ initialArticles, pageSize, cursor }) {
+  const currentCursor = ref(cursor);
+  const articlePages = ref(initialArticles ? [initialArticles] : []);
+  const fetching = ref(false);
+  const currentPage = ref(0);
+  const totalCount = ref(0);
+  const error = ref();
+
+  function onPageChange(page) {
+    currentPage.value = page;
+  }
+
+  watch(
+    [currentPage],
+    async () => {
+      if (articlePages.value[currentPage.value]) return;
+
+      fetching.value = true;
+      try {
+        const {
+          totalCount: _totalCount,
+          data: newArticles,
+          cursor: newCursor,
+        } = await $fetch(`/api/articles/`, {
```

Comment:
Please wrap this in [`useAsyncData`](https://nuxt.com/docs/api/composables/use-async-data) so the result of this call is sent to the client after SSR, as opposed to displaying the loading state and refetching on the client. 

You can see it happening when the index page is loaded with Javascript disabled:
![image](https://github.com/pantheon-systems/pantheon-content-cloud-sdk/assets/87580113/ba59b9d2-c798-4b9e-946e-016e46004664)